packages/hawtio/src/plugins/jmx/JmxContent.tsx

```diff
@@ -84,7 +87,17 @@ export const JmxContent: React.FunctionComponent = () => {
             </Routes>
           </React.Fragment>
         )}
-        {navItems.length === 0 && !selectedNode.objectName && <JmxContentMBeans />}
+        {navItems.length === 0 && !selectedNode.objectName && mBeanCollectionApplicable(selectedNode) && (
+          <AttributeTable />
+        )}
+        {navItems.length === 0 &&
+          !selectedNode.objectName &&
+          !mBeanCollectionApplicable(selectedNode) &&
+          !selectedNode.parent && <NodeNameTable />}
+        {navItems.length === 0 &&
+          !selectedNode.objectName &&
+          !mBeanCollectionApplicable(selectedNode) &&
+          selectedNode.parent && <JmxContentMBeans />}
```

Comment:
The conditionals are getting complex. I think React fragment is not a good place to embed complex conditionals like this. Maybe it's better to put them outside of the fragment and make it clearer conditionals to choose a React element.

Reply:
I think I can do that

Reply:
Fixed